### PR TITLE
Bump minimum Airflow version in providers to Airflow 2.6.0

### PR DESCRIPTION
**Pull request by eladkal**
Note: suspended providers are not included in bump min airflow version (`daskexecutor` provider)

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Review by josh-fell — APPROVED** [review]

**Comment by potiuk** [comment]
Interesting error with providers. It looks like (Can't remember that) Airflow 2.6.0 had "apache-airflow-providers-hive" dependency instead of "apache-airflow-providers-apache-hive". But it seems that 2.6.1 has it fixed already - so just setting

```
--use-airflow-version 2.6.1
```

should fix the issue.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -829,15 +829,15 @@ jobs:
           providers = json.loads(Path("generated/provider_dependencies.json").read_text())
           provider_keys = ",".join(providers.keys())
           print("AIRFLOW_EXTRAS={}".format(provider_keys))' >> $GITHUB_ENV
-      - name: "Install and verify all provider packages and airflow on Airflow 2.5 files"
+      - name: "Install and verify all provider packages and airflow on Airflow 2.6 files"
         run: >
-          breeze release-management verify-provider-packages --use-airflow-version 2.5.0
-          --use-packages-from-dist --airflow-constraints-reference constraints-2.5.0
+          breeze release-management verify-provider-packages --use-airflow-version 2.6.0
```
Comment (potiuk) [inline .github/workflows/ci.yml]:
```suggestion
          breeze release-management verify-provider-packages --use-airflow-version 2.6.1
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -829,15 +829,15 @@ jobs:
...
+          --use-packages-from-dist --airflow-constraints-reference constraints-2.6.0
```
Comment (potiuk) [inline .github/workflows/ci.yml]:
```suggestion
          --use-packages-from-dist --airflow-constraints-reference constraints-2.6.1
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -829,15 +829,15 @@ jobs:
...
         if: needs.build-info.outputs.affected-providers-list-as-string == ''
-      - name: "Install affected provider packages and airflow on Airflow 2.5 files"
+      - name: "Install affected provider packages and airflow on Airflow 2.6 files"
         run: >
-          breeze release-management install-provider-packages --use-airflow-version 2.5.0
-          --airflow-constraints-reference constraints-2.5.0 --run-in-parallel
+          breeze release-management install-provider-packages --use-airflow-version 2.6.0
```
Comment (potiuk) [inline .github/workflows/ci.yml]:
```suggestion
          breeze release-management install-provider-packages --use-airflow-version 2.6.1
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -829,15 +829,15 @@ jobs:
...
+          --airflow-constraints-reference constraints-2.6.0 --run-in-parallel
```
Comment (potiuk) [inline .github/workflows/ci.yml]:
```suggestion
          --airflow-constraints-reference constraints-2.6.1 --run-in-parallel
```

**Comment by potiuk** [comment]
Ok. Two fixes are needed:

1)           rm -rf dist/apache_airflow_providers_cohere*.whl - added here: https://github.com/apache/airflow/blob/41f4766d5b4873ddbf8daa94d837398342aeaf98/.github/workflows/ci.yml#L823

With the comment that is should be removed when we bump to 2.7.0

2) add min_airflow_version = 2.7.1 for cohere provider - seems that importlib limitation in airflow (<5) is incompatible with cohere's (>=6) - and it was only removed in 2.7.1 (https://github.com/apache/airflow/pull/3357)  

So looks ike Cohere provider is only actually installable for Airflow 2.7.1+ - we should add in their provider.yaml

**Comment by eladkal** [comment]
> add min_airflow_version = 2.7.1 for cohere provider - seems that importlib limitation in airflow (<5) is incompatible with cohere's (>=6) - and it was only removed in 2.7.1 (https://github.com/apache/airflow/pull/3357)

I think the PR you linked is wrong?
Why do you claim that cohere provider needs Airflow>=2.7.1?
I don't see anything binding that provider to newer core versions (unlike `common.fs`)

**Comment by potiuk** [comment]
> I think the PR you linked is wrong? Why do you claim that cohere provider needs Airflow>=2.7.1? I don't see anything binding that provider to newer core versions (unlike `common.fs`)

Yep. Nine was removed when copy*pasting (https://github.com/apache/airflow/pull/33579)

**Review by Taragolis — APPROVED** [review]
Looks good. There are still some backcompat code presented, but it quite difficult to remove it in one go, so it could be remove as follow up

**Comment by eladkal** [comment]
FYI @kaxil @ephraimbuddy @utkarsharma2

`ERROR: Cannot install apache-airflow-providers-cohere, apache-airflow==2.6.1 and cohere==4.37 because these package versions have conflicting dependencies.`

The cohere provider can't work with 2.6 it needs min airflow version of 2.7.1 due to version of `importlib`

**Comment by eladkal** [comment]
> The cohere provider can't work with Airflow 2.6 unless using Python 3.9 due to version of importlib

OK thanks to @potiuk we may have workaround for this

https://github.com/apache/airflow/pull/36017/commits/8ad4e9bbb2c46f907718f5c417b01e4da232fea5

**Comment by eladkal** [comment]
rebased after https://github.com/apache/airflow/pull/36061
lets see if CI passes

**Comment by potiuk** [comment]
> rebased after #36061 lets see if CI passes

Close enough - you need to also remove `2.5.0` from the list : 

https://github.com/apache/airflow/blob/main/dev/README_RELEASE_PROVIDER_PACKAGES.md#bump-min-airflow-version-for-providers

> Update BASE_PROVIDERS_COMPATIBILITY_CHECKS in src/airflow_breeze/global_constants.py to remove the versions of Airflow that are not applicable any more.

**Comment by eladkal** [comment]
> Close enough - you need to also remove 2.5.0 from the list 

Thanks! I new i forgot something

**Comment by ephraimbuddy** [comment]
This should be in 2.8 right @eladkal ?

**Comment by eladkal** [comment]
> This should be in 2.8 right @eladkal ?

If the breeze changes are relevant yes.

**Comment by potiuk** [comment]
I think it's not **strictly** necessary for 2.8 - I will cherry-pick it eventually anyway (if @ephraimbuddy will not do it ) - but for 2.8.0 (i will eventually make it completely uneccessary when we move to more modern tooling for building airflow package ~ couple of months or so.

**Pull request merged**